src/poetry/console/commands/sync.py

```diff
@@ -0,0 +1,36 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+from typing import ClassVar
+
+from poetry.console.commands.install import InstallCommand
+
+
+if TYPE_CHECKING:
+    from cleo.io.inputs.option import Option
+
+
+class SyncCommand(InstallCommand):
+    name = "sync"
+    description = "Update the project's environment according to the lockfile."
+
+    options: ClassVar[list[Option]] = [
+        opt for opt in InstallCommand.options if opt.name != "sync"
```

Comment:
I want to avoid _this_ breaking change because it would break a lot of CI scripts. (Most of our other breaking changes are not really relevant for most users or at least not relevant for CI.) But we can deprecate it.

Reply:
ok, let's deprecate it, but with a set date (June 2025 sounds reasonable). I'd like us to avoid a long-living deprecation we are stuck with forever.

Reply:
Hmm, there is also `poetry self install --sync`.

Option a) Do not deprecate `--sync` for `poetry self install` (only for `poetry install`).
Option b) Introduce `poetry self sync` for the symmetry.